tests/rl/test_health_monitor.py

```diff
@@ -0,0 +1,360 @@
+            assert len(warning_calls) > 0
+
+    def test_alert_does_not_trigger_on_stable_performance(self):
+        """Test that alert doesn't trigger for stable performance."""
+        curriculum = create_test_curriculum_with_alerts(
+            [PerformanceVolatilityAlert(volatility_threshold=0.2, window=30)]
+        )
+
+        with patch("marin.rl.curriculum.logger") as mock_logger:
+            # Stable performance
+            for step in range(50):
+                rewards = [0.5] * 5  # Consistent performance
+                rollout_stats = [create_test_rollout_stats(r, "lesson1") for r in rewards]
+                curriculum.update_lesson_stats(rollout_stats, "training", step)
+
+            # Should not trigger volatility alert
+            warning_calls = [call for call in mock_logger.warning.call_args_list if "volatility" in str(call)]
+            assert len(warning_calls) == 0
+
+
+class TestAlertIntegration:
+    """Test alert system integration with curriculum."""
+
+    def test_multiple_alerts_on_same_lesson(self):
+        """Test that multiple alerts can be configured for one lesson."""
+        curriculum = create_test_curriculum_with_alerts(
+            [
+                TrainingStalledAlert(stagnation_window=30),
+                PerformanceVolatilityAlert(volatility_threshold=0.2),
+            ]
+        )
+
+        # Test volatility alert triggers
+        with patch("marin.rl.curriculum.logger") as mock_logger:
+            curriculum._last_alert_time.clear()
+            # Volatile performance
+            for step in range(50):
+                perf = 0.5 if step % 2 == 0 else 0.3  # Volatile
+                rewards = [perf] * 5
+                rollout_stats = [create_test_rollout_stats(r, "lesson1") for r in rewards]
+                curriculum.update_lesson_stats(rollout_stats, "training", step)
+
+            # Volatility alert should trigger
+            volatility_calls = [call for call in mock_logger.warning.call_args_list if "volatility" in str(call)]
+            assert len(volatility_calls) > 0
+
+        # Test stagnation alert triggers (separate scenario)
+        curriculum2 = create_test_curriculum_with_alerts([TrainingStalledAlert(stagnation_window=30, plateau_window=20)])
+        with patch("marin.rl.curriculum.logger") as mock_logger:
+            curriculum2._last_alert_time.clear()
+            # Flat performance (plateaued)
+            for step in range(50):
+                rewards = [0.5] * 5  # Flat
+                rollout_stats = [create_test_rollout_stats(r, "lesson1") for r in rewards]
+                curriculum2.update_lesson_stats(rollout_stats, "training", step)
+
+            # Stagnation alert should trigger
+            stagnation_calls = [call for call in mock_logger.warning.call_args_list if "no progress" in str(call)]
+            assert len(stagnation_calls) > 0
+
+    def test_alerts_only_evaluate_when_conditions_met(self):
```

Comment:
nit: this is up to you, but I'd probably focus on the wandb interactions, since those are the ones we're going to pay the most attention to. logger based tests just tend to be finicky, but again, not a big deal